### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by francksefu**
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
In your PR description include the screenshots of your app's logs before and after the fix
This is the corresponding wireframe view: blog all user posts
Integration specs
Use Capybara to write integration tests for each view in your project.

Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

Before :
Capture d’écran 2024-01-24 à 2 06 04 AM
After : 
Capture d’écran 2024-01-24 à 2 08 31 AM

### CODE REVIEW

File: spec/views_test/user_show_spec.rb
Comment (IngDavidTM) [inline spec/views_test/user_show_spec.rb:1]: - [ ] Although you performed exceptionally well, you overlooked checking the user's post count. Kindly include that test :+1:.

File: spec/views_test/post_index_spec.rb
Comment (IngDavidTM) [inline spec/views_test/post_index_spec.rb:1]: - [ ] Great work in developing tests for your index post views. However, you overlooked incorporating tests for scenarios involving pagination when there are more posts than can fit on the view, tracking the number of likes per post, and monitoring the number of comments per post. Please incorporate these tests. 👍

File: spec/views_test/post_show_spec.rb
Comment (IngDavidTM) [inline spec/views_test/post_show_spec.rb:1]: - [ ] Excellent work in crafting tests for your show post views. However, you omitted to include tests that display the post author and the usernames of each commenter. Kindly integrate these tests.

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi Team 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] Well done on the tests, but every time I attempt to execute them, they fail. Please verify and ensure that they are functioning as intended.

Screenshot from 2024-01-23 19-21-55

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — APPROVED** [review]
### Hi @francksefu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**